eclair-core/src/main/scala/fr/acinq/eclair/router/Announcements.scala

```diff
@@ -89,6 +91,17 @@ object Announcements {
     )
   }
 
+  case class AddressException(message: String) extends IllegalArgumentException(message)
+
+  def validateAddresses(addresses: List[NodeAddress]): Option[AddressException] = {
```

Comment:
I was originally thinking about it as something related to the Announce specification, but agree now since it's only used when parsing NodeParams it should be moved.

Fixed in [34e6bef](https://github.com/ACINQ/eclair/pull/2234/commits/34e6befa476f84efec66559567b7e6442bf23c65).